File: abc137_d.py
# https://atcoder.jp/contests/abc137/tasks/abc137_d

import heapq
N, M = map(int, input().split())

days = [[] for _ in range(M+1)]
for i in range(N):
    A, B = map(int, input().split())
    if A <= M:
        days[A].append(B)

ans = 0
q = []
for j in range(1, M+1):
    for k in days[j]:
        heapq.heappush(q, -k)

    if q:
        tmp = heapq.heappop(q)
        ans -= tmp
    
print(ans)

File: atcoder/abc137_d.cpp
// https://atcoder.jp/contests/abc137/tasks/abc137_d
#include<iostream>
#include<algorithm>
//#include <numeric>
//#include<string>
#include<vector>
//#include<map>
//#include<tuple>
#include<queue>
//#include<regex>

// #include <bits/stdc++.h>
#define rep(i,n) for (int i = 0; i < (n); ++i)
using namespace std;
typedef long long ll;

int main() {
	int n, m;
	cin >> n >> m;
	vector<vector<int>> jobs(m + 1);
	rep(i, n) {
		int a, b;
		cin >> a >> b;
		if (a > m) continue;
		jobs[m - a].push_back(b);
	}
	priority_queue<int> q;
	ll ans = 0;
	for (int i = m-1; i >= 0; --i)	{
		for (int b: jobs[i])		{
			q.push(b);
		}
		if (!q.empty())		{
			ans += q.top();
			q.pop();
		}
	}

	cout << ans << endl;

	return 0;
}
